dist/src/main/java/io/camunda/zeebe/shared/security/IdentityAuthenticationManager.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.shared.security;
+
+import io.camunda.identity.sdk.Identity;
+import io.camunda.zeebe.protocol.record.value.TenantOwned;
+import java.util.Collections;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.ReactiveAuthenticationManager;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Component;
+import reactor.core.publisher.Mono;
+
+@Component
+public final class IdentityAuthenticationManager implements ReactiveAuthenticationManager {
+
+  private final Identity identity;
+
+  @Autowired
+  public IdentityAuthenticationManager(final Identity identity) {
+    this.identity = identity;
+  }
+
+  @Override
+  public Mono<Authentication> authenticate(final Authentication authentication) {
+    if (!(authentication instanceof PreAuthToken preAuthToken)) {
+      return Mono.just(authentication);
+    }
+
+    final List<String> tenants;
+    final var tokenValue = preAuthToken.token();
+
+    try {
+      final var token = identity.authentication().verifyToken(tokenValue);
+      tenants = Collections.singletonList(TenantOwned.DEFAULT_TENANT_IDENTIFIER);
```

Comment:
:thought_balloon: I guess you'll integrate multi-tenancy here.

Reply:
Exactly, and then we can add the QA tests too :+1: